flink-runtime-web/web-dashboard/src/app/share/common/dagre/svg-container.component.ts

```diff
@@ -94,12 +86,12 @@ export class SvgContainerComponent implements OnInit, AfterContentInit {
           this.transform = `translate(${this.containerTransform.x} ,${this.containerTransform.y})scale(${this.containerTransform.k})`;
         }
         this.zoomEvent.emit(this.zoom);
-        this.transformEvent.emit(this.containerTransform as SafeAny);
+        this.transformEvent.emit((this.containerTransform as unknown) as { x: number; y: number; scale: number });
```

Comment:
Why the weird double casting? I looked this one up as an example (since there were so many cases), and `containerTransform` has properties `x`, `y`, `k`. The latter might mean the same as `scale`, but here `scale` will just be undefined, right?

Reply:
Yes, so there is some really weird type mismatch going on between `k` and `scale` being used. You cannot type-assert this directly, the compiler won't let you because there is not enough overlap for that to ever work. The trick to assert it to `unknown` first is recommended by the compiler itself. This is definitely a symptom of something being broken, though.

Reply:
Ok, I was just confused by the weird type mismatches going on, but as you said - let's fix it some other time